### DISCUSSION TITLE
How to inset markdown into xxx.stories.tsx ?

### DISCUSSION THREAD

**Discussion by Flcwl** [storybookjs/storybook#23183] [Help] [answered]
How to inset markdown into xxx.stories.tsx ?  Storybook V7

**Comment by f3bw** [comment, marked as answer]
To insert Markdown into a `xxx.stories.tsx` file in Storybook V7, you can use the `mdx` addon provided by Storybook. Follow these steps:

1. Install the necessary dependencies:
   ```bash
   npm install @storybook/addon-docs
   ```

2. Create or open the `preview.js` file in your Storybook configuration folder (typically `.storybook`). If the file doesn't exist, create it.

3. Import the `addDecorator` function from `@storybook/addon-docs` in your `preview.js` file:
   ```jsx
   import { addDecorator } from '@storybook/addon-docs';
   ```

4. Add the `addDecorator` function to wrap your stories with the `mdx` decorator:
   ```jsx
   addDecorator((Story) => (
     <>
       <h1>Your Custom Heading</h1>
       <Story />
     </>
   ));
   ```

   This code adds a custom heading before each story. You can modify it to suit your needs.

5. In your `xxx.stories.tsx` file, import and use the `markdown` loader from Storybook:
   ```jsx
   import { withMarkdown } from 'storybook-addon-markdown';

   export default {
     title: 'Your Component',
     decorators: [withMarkdown],
   };
   ```

6. Now you can include Markdown content directly in your story using the `.md` file extension. For example, create a `YourComponent.stories.md` file and write your Markdown content in it:
   ```markdown
   ## Your Markdown Content

   Some text and *markdown formatting* goes here.
   ```

   When you run Storybook, it will load the Markdown content and display it alongside your component.

Note: The `withMarkdown` function is provided by the `@storybook/addon-docs` package, which was previously included in Storybook V6 by default but became an add-on in Storybook V7. If you haven't upgraded from Storybook V6 or if you're starting a new project with Storybook V7, you may need to install `@storybook/addon-docs` manually as shown in step 1.
